### DISCUSSION TITLE
Build error...

### DISCUSSION THREAD

**Discussion by Bunker56** [PhlexPlexico/G5API#202] [Q&A] [answered]
I tried it with version 16.5.0 and 16.15.0 - the result is the same. The assembly of the G5V passed without problems. Am I doing something wrong or is there something wrong?

g5api@bunker:~/G5API$ yarn
yarn install v1.22.19
[1/6] Validating package.json...
[2/6] Resolving packages...
[3/6] Fetching packages...
[4/6] Linking dependencies...
warning "swagger-jsdoc > swagger-parser > @apidevtools/swagger-parser@10.0.2" has unmet peer dependency "openapi-types@>=7".
warning Workspaces can only be enabled in private projects.
[5/6] Building fresh packages...
[1/4] ⠁ bcrypt
[2/4] ⠁ cpu-features
[-/4] ⠁ waiting...
warning Error running install script for optional dependency: "/home/g5api/G5API/node_modules/cpu-features: Command failed.
Exit code: 1
Command: node-gyp rebuild
Arguments: 
Directory: /home/g5api/G5API/node_modules/cpu-features
Output:
gyp info it worked if it ends with ok
gyp info using node-gyp@9.0.0
gyp info using node@16.15.0 | linux | x64
gyp info find Python using Python version 3.9.2 found at \"/usr/bin/python3\"
gyp info spawn /usr/bin/python3
gyp info spawn args [
gyp info spawn args   '/home/g5api/.nvm/versions/node/v16.15.0/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/home/g5api/G5API/node_modules/cpu-features/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/home/g5api/.nvm/versions/node/v16.15.0/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/home/g5api/.cache/node-gyp/16.15.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/home/g5api/.cache/node-gyp/16.15.0',
gyp info spawn args   '-Dnode_gyp_dir=/home/g5api/.nvm/versions/node/v16.15.0/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/home/g5api/.cache/node-gyp/16.15.0/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/home/g5api/G5API/node_modules/cpu-features',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp ERR! build error 
gyp ERR! stack Error: not found: make
gyp ERR! stack     at getNotFoundError (/home/g5api/.nvm/versions/node/v16.15.0/lib/node_modules/npm/node_modules/which/which.js:10:17)
gyp ERR! stack     at /home/g5api/.nvm/versions/node/v16.15.0/lib/node_modules/npm/node_modules/which/which.js:57:18
gyp ERR! stack     at new Promise (<anonymous>)
gyp ERR! stack     at step (/home/g5api/.nvm/versions/node/v16.15.0/lib/node_modules/npm/node_modules/which/which.js:54:21)
gyp ERR! stack     at /home/g5api/.nvm/versions/node/v16.15.0/lib/node_modules/npm/node_modules/which/which.js:71:22
gyp ERR! stack     at new Promise (<anonymous>)
gyp ERR! stack     at subStep (/home/g5api/.nvm/versions/node/v16.15.0/lib/node_modules/npm/node_modules/which/which.js:69:33)
gyp ERR! stack     at /home/g5api/.nvm/versions/node/v16.15.0/lib/node_modules/npm/node_modules/which/which.js:80:22
gyp ERR! stack     at /home/g5api/.nvm/versions/node/v16.15.0/lib/node_modules/npm/node_modules/isexe/index.js:42:5
gyp ERR! stack     at /home/g5api/.nvm/versions/node/v16.15.0/lib/node_modules/npm/node_modules/isexe/mode.js:8:5
gyp ERR! System Linux 5.10.0-14-amd64
gyp ERR! command \"/home/g5api/.nvm/versions/node/v16.15.0/bin/node\" \"/home/g5api/.nvm/versions/node/v16.15.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js\" \"rebuild\"
gyp ERR! cwd /home/g5api/G5API/node_modules/cpu-features
gyp ERR! node -v v16.15.0
[6/6] Cleaning modules...
Done in 6.28s.

g5api@bunker:~/G5V$ nvm -v
0.39.1
g5api@bunker:~/G5V$ npm -v
8.11.0
g5api@bunker:~/G5V$ node -v
v16.15.0
g5api@bunker:~/G5V$ yarn -v
1.22.19
g5api@bunker:~/G5V$

**Comment by PhlexPlexico** [comment, marked as answer]
What architecture are you using? ARM, or x86/64?

**Reply by PhlexPlexico** [reply]
This usually means you're missing some build tools for bcrypt, which encrypts users passwords for logging in. I would make sure you have all the dependencies installed for this on your machine. Try `sudo apt install make` first, and see if that's all you need :)

**Reply by Bunker56** [reply]
Just a moment...

**Comment by Bunker56** [comment]
x86/64

**Reply by PhlexPlexico** [reply]
`gyp ERR! stack Error: not found: make`

Make sure `make` is installed :)

**Reply by Bunker56** [reply]
Thanks for the hint, I forgot about dependencies...

**Reply by PhlexPlexico** [reply]
No worries! A full list of deps can be found in the Dockerfile as well, if it's still failing!

**Reply by Bunker56** [reply]
Thanks! It's all right. Now we will fight with the settings ... :)

**Reply by Bunker56** [reply]
One more question.
redis-server /etc/redis/redis.conf --daemonize yes --appendonly yes
Is it mandatory or not?

**Reply by PhlexPlexico** [reply]
Yes, as all sessions are logged in redis.

**Reply by Bunker56** [reply]
Thnx!..

**Comment by Bunker56** [comment]
Today there was such an error...
Repository error?

-----                                                                                                                                     
 > [2/7] RUN apk add gettext python3 build-base:                                                                                           
#0 0.599 fetch https://dl-cdn.alpinelinux.org/alpine/v3.17/main/x86_64/APKINDEX.tar.gz                                                     
#0 5.602 ERROR: https://dl-cdn.alpinelinux.org/alpine/v3.17/main: temporary error (try again later)                                        
#0 5.602 WARNING: Ignoring https://dl-cdn.alpinelinux.org/alpine/v3.17/main: No such file or directory                                     
#0 5.602 fetch https://dl-cdn.alpinelinux.org/alpine/v3.17/community/x86_64/APKINDEX.tar.gz                                                
#0 10.61 ERROR: https://dl-cdn.alpinelinux.org/alpine/v3.17/community: temporary error (try again later)
#0 10.61 WARNING: Ignoring https://dl-cdn.alpinelinux.org/alpine/v3.17/community: No such file or directory
#0 10.61 ERROR: unable to select packages:
#0 10.61   build-base (no such package):
#0 10.61     required by: world[build-base]
#0 10.61   gettext (no such package):
#0 10.61     required by: world[gettext]
#0 10.61   python3 (no such package):
#0 10.61     required by: world[python3]

**Comment by Bunker56** [comment]
I'm sorry, there seems to be a problem with the firewall...
